text-command-line-terminal-applications-and-interfaces.md

```diff
@@ -1,8 +1,25 @@
 Text / Command-line / Terminal Applications and Interfaces
 ----------------------------------------------------------
 
-Text applications are a class of software ICT that appeared decades ago, prior to the emergence of the graphical user interface (GUI) and the Web. The interface of a text application is generated using only text characters, and either a hardware terminal or a software terminal application handles the rendering of the text application—similar to how a web user agent handles the rendering of a web application. Text applications only accept text input (though some terminal applications which render text applications in the GUI may utilize a mouse or other input devices). Command-line applications are a subset of text applications with further specific properties.
+Text applications are a class of software ICT that appeared decades ago, prior to the emergence of the graphical user interface (GUI) and the Web. The interface of a text application is generated using only text characters, and either a hardware terminal or a software terminal application handles the rendering of the text application—similar to how a web user agent handles the rendering of a web application. Text applications only accept text input, though some may also support the use of a mouse. More recently, terminal applications which render text applications in the GUI may utilize spoken input through Automated Speech Recognition (ASR) (or other input devices). Both GUI and native text environment interfaces also now commonly support word-completion prediction technologies. Command-line applications are a subset of text applications with further specific properties.
```

Comment:
The meaning was changed in this sentence from what it used to be: 
>Previous version: Text applications only accept text input (though some terminal applications which render text applications in the GUI may utilize a mouse or other input devices).

>New version: Text applications only accept text input, though some may also support the use of a mouse. 

My thinking is that if a text application is intended to be accessed through a GUI Terminal Emulator, then mouse interaction could be possible. If it's not rendered through a GUI emulator, then it is only usable via keyboard. Is my understanding accurate? 

Perhaps this is clearer:

Text applications rendered in a text-only terminal application will only accept text input. When rendered through a GUI terminal application, it could also support the use of a mouse. More recently, terminal applications which render text applications in the GUI may also support spoken input through Automated Speech Recognition (ASR) (or other input devices).

Reply:
No change needed, ignore based on our conversation.